### PR TITLE
Add support for Scala 3

### PR DESCRIPTION
**Pull request by ingarabr**
- Add scala 3 cross build
- Reimplemented macros for scala 3
- Upgrade typelevel dependencies and fixed deprecation warnings
- Use Typelevel literals ( @hamnis )

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,37 +21,60 @@ lazy val root = (project in file("."))
...
     scalaVersion := "2.12.12",
-    crossScalaVersions := Seq("2.12.12", "2.13.6"),
-    scalacOptions ++= Seq(
-      "-language:experimental.macros",
-      "-Yrangepos",
-      "-Ywarn-unused",
-      "-Xlint",
-      "-P:semanticdb:synthetics:on"
-    ),
+    crossScalaVersions := Seq("2.12.12", "2.13.6", "3.2.2"),
```
Comment (hamnis) [inline build.sbt:24]:
```suggestion
    crossScalaVersions := Seq("2.12.17", "2.13.10", "3.2.2"),
```

File: src/main/scala-2/com/minosiants/pencil/protocol/ScodecCompat.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.minosiants.pencil.protocol
+
+import scodec.bits.BitVector
+import scodec.{Attempt, Codec}
+
+trait ScodecCompat {
+  implicit class ScodecCodecCompat[A](codec: Codec[A]) {
+    def encodeAll(values: List  [A]): Attempt[BitVector] = {
```
Comment (hamnis) [inline src/main/scala-2/com/minosiants/pencil/protocol/ScodecCompat.scala:8]:
```suggestion
    def encodeAll(values: List[A]): Attempt[BitVector] = {
```

File: src/test/scala/com/minosiants/pencil/SmtpServer.scala [hunk 3]
```diff
@@ -40,43 +40,43 @@ final case class SmtpServer(
 
     stream.flatMap {
       case In(raw, Ehlo(_)) =>
-        Stream.eval_(state.tryUpdate(_ :+ raw)) ++
+        Stream.eval(state.tryUpdate(_ :+ raw)).drain ++
```
Comment (hamnis) [inline src/test/scala/com/minosiants/pencil/SmtpServer.scala:43]: `Stream.exec` without drain
assuming `state.tryUpdate` returns a `F[Unit]`

Comment (ingarabr) [reply src/test/scala/com/minosiants/pencil/SmtpServer.scala:43]: It returns a boolean.

**Comment by minosiants** [comment]
Thanks you guys for doing this !

**Comment by ingarabr** [comment]
Happy to help out.

**Pull request merged**